### PR TITLE
[FEATURE] Better debug messages and values.

### PR DESCRIPTION
**Pull request by jpfeuffer**
# Description

Provides a little bit easier debugging in PLFQ.

### CODE REVIEW

File: src/openms/source/ANALYSIS/OPENSWATH/MRMFeatureFinderScoring.cpp [hunk 1]
```diff
@@ -1129,6 +1129,10 @@ namespace OpenMS
       }
       chromatogram.setMetaValue("product_mz", transition->getProductMZ());
       chromatogram.setMetaValue("precursor_mz", transition->getPrecursorMZ());
+      Precursor prec; prec.setPosition(transition->getPrecursorMZ());
```
Comment (jpfeuffer) [inline src/openms/source/ANALYSIS/OPENSWATH/MRMFeatureFinderScoring.cpp:1132]: I wonder if that breaks the SWATH tests. But It looks like the metaValues are legacy anyways. Some debug messages indirectly depend on the Precursor being set.

Comment (jpfeuffer) [reply src/openms/source/ANALYSIS/OPENSWATH/MRMFeatureFinderScoring.cpp:1132]: Ok have to check if the failing tests are just cosmetic.

**Comment by jpfeuffer** [comment]
How do we store UInt64 in Metavalues? Can we? @timosachsenberg 
See ProteomicsLFQ3 failing test.

**Comment by jpfeuffer** [comment]
rebuild jenkins

File: src/openms/source/ANALYSIS/OPENSWATH/MRMFeatureFinderScoring.cpp (hunk 1 above)
Comment (jpfeuffer) [reply src/openms/source/ANALYSIS/OPENSWATH/MRMFeatureFinderScoring.cpp:1132]: @cbielow @hroest Any comments on this. Feels like it should be okay. Tests are running. Are the Precursor and Product fields the "new" alternative to the metavalues?

**Comment by jpfeuffer** [comment]
@cbielow @oliveralka I am interested in seeing, where certain Features, Transitions, Chromatogram are getting filtered out due to bad fits, missing apices, bad alignment quality.
Do you have any inputs on where to put additional debug statements to find this out?

File: src/openms/source/TRANSFORMATIONS/FEATUREFINDER/FeatureFinderIdentificationAlgorithm.cpp [hunk 2]
```diff
@@ -665,7 +667,12 @@ namespace OpenMS
             // to not be combined
             seedcount++;
             //cout << peptide.sequence << " " << charge << endl;
-            String peptide_id = peptide.sequence + String(seedcount) + "/" + String(charge);
+            double mz = rt_pep.second->getMZ();
+            double rt = rt_pep.second->getRT();
+            String uid = rt_pep.second->getMetaValue("SeedFeatureID");
+
+            // UID should be enough, but let's add the seed count to be sure.
+            String peptide_id = peptide.sequence + "[" + uid + "][" + String(seedcount) + "]/" + String(charge);
```
Comment (cbielow) [inline src/openms/source/TRANSFORMATIONS/FEATUREFINDER/FeatureFinderIdentificationAlgorithm.cpp:675]: why not directly on `peptide.id = ...` ?

**Review by cbielow — APPROVED** [review]
looks good

File: src/openms/source/ANALYSIS/OPENSWATH/MRMFeatureFinderScoring.cpp (hunk 1 above)
Comment (timosachsenberg) [inline src/openms/source/ANALYSIS/OPENSWATH/MRMFeatureFinderScoring.cpp:1132]: looks a bit strange to have both expressions in one line. below too

File: src/openms/source/TRANSFORMATIONS/FEATUREFINDER/FeatureFinderIdentificationAlgorithm.cpp [hunk 3]
```diff
@@ -336,6 +336,8 @@ namespace OpenMS
         peptides.back().setRT(f_it->getRT());
         peptides.back().setMZ(f_it->getMZ());
         peptides.back().setMetaValue("FFId_category", "internal");
+        peptides.back().setMetaValue("FromUntargeted", "true");
```
Comment (timosachsenberg) [inline src/openms/source/TRANSFORMATIONS/FEATUREFINDER/FeatureFinderIdentificationAlgorithm.cpp:339]: do we need this? if there is a seed we should know that it is form untargeted?

**Review by timosachsenberg — APPROVED** [review]

File: src/openms/source/TRANSFORMATIONS/FEATUREFINDER/FeatureFinderIdentificationAlgorithm.cpp (hunk 3 above)
Comment (jpfeuffer) [reply src/openms/source/TRANSFORMATIONS/FEATUREFINDER/FeatureFinderIdentificationAlgorithm.cpp:339]: no but then I have to redo all tests

Comment (timosachsenberg) [reply src/openms/source/TRANSFORMATIONS/FEATUREFINDER/FeatureFinderIdentificationAlgorithm.cpp:339]: Also, we never set FromUntargeted to "false" for the other features.
Maybe I am picky here but like this it doesn't really add value but just needs extra memory.

Comment (timosachsenberg) [reply src/openms/source/TRANSFORMATIONS/FEATUREFINDER/FeatureFinderIdentificationAlgorithm.cpp:339]: I think we need a script that updates all failed tests ;)

Comment (timosachsenberg) [reply src/openms/source/TRANSFORMATIONS/FEATUREFINDER/FeatureFinderIdentificationAlgorithm.cpp:339]: ok I will update them

**Pull request merged**